src/leaderboard/leaderboard.service.ts

```diff
@@ -22,47 +20,66 @@ export class LeaderboardService {
     return this.leaderboardRepository.save(newLeaderboard);
   }
 
-  async sortLeaderboard(
-    leaderboardId: string,
-    leaderboardName: string,
-    metrics: string[],
-    sortingOrder: SortingOrders[],
-    index: number,
-    list: PlayerLeaderboard[] | undefined,
-  ): Promise<SortedResult> {
-    if (list === undefined) {
-      const list = await this.playerLeaderboardRepository.find({
-        where: {
-          leaderboardId: leaderboardId,
-        },
-      });
-    }
+  async sortLeaderboard(leaderboardId: string): Promise<any> {
+    let temp = 0;
+    let start = 0;
+    let end = 0;
+    let duplicatesIndex = 0;
+    let preSorted = [];
+    let postSorted = [];
 
-    if (sortingOrder[index].includes(SortingOrders.ASC)) {
-      list.sort((x, y) => (x.score[metrics[index]] > y.score[metrics[index]] ? -1 : 1));
-    }
-    if (sortingOrder[index].includes(SortingOrders.DESC)) {
-      list.sort((x, y) => (x.score[metrics[index]] > y.score[metrics[index]] ? 1 : -1));
-    }
+    const leaderboard: Leaderboard = await this.leaderboardRepository.findOne(leaderboardId);
+    let list = await this.playerLeaderboardRepository.find({
+      where: {
+        leaderboardId: leaderboardId,
+      },
+    });
 
-    const tempScores = [];
+    for (let i = 0; i < leaderboard.metrics.length; i++) {
+      let outcome = 1;
+      const metric = leaderboard.metrics[i];
+      const sortingOrder = leaderboard.sortingOrders[i];
+      const reverse = sortingOrder === SortingOrders.DESC ? -1 : 1;
 
-    for (const player of list) {
-      tempScores.push(player.score[metrics[index]]);
-    }
+      list.sort((a, b) => {
+        if (a.score[metric] < b.score[metric]) {
+          return reverse * -1;
+        } else if (a.score[metric] > b.score[metric]) {
+          return reverse * 1;
+        } else {
+          temp = a.score[metric];
+          duplicatesIndex++;
```

Comment:
if an array contains duplicates, we will see it when the this index is different than 0 basically